Source/Tools/LinuxEmulation/LinuxSyscalls/ThreadManager.h

```diff
@@ -84,6 +84,14 @@ struct ThreadStateObject : public FEXCore::Allocator::FEXAllocOperators {
   std::atomic_bool ThreadSleeping {false};
   FEXCore::InterruptableConditionVariable ThreadPaused;
 
+  // GDB signal information
+  struct GdbInfoStruct {
+    int Signal {};
+    uint64_t SignalPC {};
+    uint64_t GPRs[32];
+    uint64_t PState {};
+  } GdbInfo;
```

Comment:
When is the data here supposed to be valid / up to date? Does it make sense to turn it into an `std::optional` so we can assert against "overwritten while still in use" or "not initialized" errors?

Reply:
Switched to an optional.